src/hsmm_mvpy/models.py

```diff
@@ -298,16 +349,16 @@ def EM(self, n_events, magnitudes, parameters,  threshold, magnitudes_to_fix=Non
         lkh, eventprobs = self.estim_probs(magnitudes, parameters, n_events)
         traces = []
         i = 0
-        if threshold == 0 or n_events==0:
+        if not maximization or n_events==0:
             lkh_prev = lkh
             magnitudes_prev = initial_magnitudes
             parameters_prev = initial_parameters
             eventprobs_prev = eventprobs
         else:
             while i < max_iteration :#Expectation-Maximization algorithm
-                if i > min_iteration and lkh - lkh_prev < threshold:
+                if i > min_iteration and tolerance > lkh - lkh_prev:
```

Comment:
any reason for keeping min_iteration?

Reply:
Not really, do you think we should remove?

Reply:
yes, it's only confusing and takes some time to check.

Reply:
Ah now I remember why I left it. We need a first run to verify ```lkh - lkh_prev```, in the previous versions we made an extra loop to get the first ```lkh_prev```, now it's replaced by enforcing at least 1 iteration. I suggest we keep as I'll need to track where it is declared

Reply:
what about setting lkh_prev to -inf instead to force the first run?

Reply:
True that's enough, I'll make the change at some point